driver/src/price_finding/price_finder_interface.rs

```diff
@@ -0,0 +1,51 @@
+use crate::models;
+
+use crate::error::DriverError;
+use web3::types::U256;
+
+#[derive(Clone)]
+pub struct Solution {
+    pub surplus: U256,
+    pub prices: Vec<u128>,
+    pub executed_sell_amounts: Vec<u128>,
+    pub executed_buy_amounts: Vec<u128>,
```

Comment:
WHy do we need both buy and sells here?

Reply:
It's part of Tom's solution, but you are right we don't strictly need it. I can remove it.

Reply:
We do need buy and sell in our SellOrder collection on the contract. Are these related?

Notice that `maxSellAmount` and `minBuyAmount` are used to interpret the `limitPrice` of a sell order. @twalth3r can back this up if need be.

Reply:
This is just for the solutions. The smart contract will get the prices and one side of the vector. @twalth3r were these sufficient to recover the exact balances, even when we have rounding or do we need to push the epsilons as well?